File: AndroidVersion4.1.3/Budget App/app/src/main/java/com/example/budgy/LoginActivity.java
package com.example.budgy;

import androidx.appcompat.app.AppCompatActivity;

import android.app.AlertDialog;
import android.content.Intent;
        import android.os.Bundle;
import android.util.Log;
import android.view.View;
        import android.widget.Button;
        import android.widget.EditText;
        import android.widget.TextView;
import android.widget.Toast;

import com.example.budgy.Models.Budget;
import com.example.budgy.Models.LoggedInUser;
import com.example.budgy.Models.User;

import java.util.Calendar;

import static com.example.budgy.ui.settings.Settings.getMonth;

public class LoginActivity extends AppCompatActivity
{
    DatabaseHelper db;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);
        final Button btnReg = (Button)findViewById(R.id.btnReg);
        EditText edtxtUser = (EditText)findViewById(R.id.edtxtUser);
        EditText edtxtPass = (EditText)findViewById(R.id.edtxtPass);
        Button btnLogin = (Button)findViewById(R.id.btnLogin);
        TextView txtNewUser = (TextView)findViewById(R.id.txtNewUser);

        db = new DatabaseHelper(this);

        btnReg.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view)
            {
                startActivity(new Intent(LoginActivity.this, Registration.class));
            }
        });

        btnLogin.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view)
            {
                String userInput = edtxtUser.getText().toString().trim();
                String passwordInput = edtxtPass.getText().toString().trim();
                User user = db.getUserByEmailAndPassword(userInput, passwordInput);

                if (user == null)
                {
                    Toast.makeText(getApplicationContext(),"Invalid Credentials",Toast.LENGTH_LONG).show();
                }
                else
                {
                    LoggedInUser.setLoggedInUser(user);
                    String month, year;
                    month = getMonth(Calendar.getInstance().get(Calendar.MONTH));
                    year = Calendar.getInstance().get(Calendar.YEAR) + "";
                    Budget budget = new Budget(db.getCurrentBudget(LoggedInUser.getLoggedInUser().getID(), year, month));
                    LoggedInUser.setCurrentBudget(budget.getAmount());
                    if (LoggedInUser.getCurrentBudget().equals("")) {
                        db.insertBudget(month, year, "0", LoggedInUser.getLoggedInUser().getID());
                        LoggedInUser.setCurrentBudget("0");
                    }
                    else {
                        LoggedInUser.setCurrentBudget(db.getCurrentBudget(LoggedInUser.getLoggedInUser().getID(), year, month).getAmount());
                    }
                    finish();
                    startActivity(new Intent(LoginActivity.this, MainActivity.class));
                }
            }
        });

    }
}

File: 2021 Special Topic Web Mobile App/PocketBook_Version1/app/src/main/java/com/example/pocketbook/LoginActivity.kt
package com.example.pocketbook

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast
import java.util.*

class LoginActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)

        val btnReg = findViewById<View>(R.id.btnReg) as Button
        val edtxtUser = findViewById<View>(R.id.edtxtUser) as EditText
        val edtxtPass = findViewById<View>(R.id.edtxtPass) as EditText
        val btnLogin = findViewById<View>(R.id.btnLogin) as Button
        val txtNewUser = findViewById<View>(R.id.txtNewUser) as TextView

        btnReg.setOnClickListener {
            startActivity(Intent(this@LoginActivity, RegistrationActivity::class.java)
            )
        }

        btnLogin.setOnClickListener {

            startActivity(Intent(this@LoginActivity, NavigationDrawerActivity::class.java))

//            val userInput = edtxtUser.text.toString().trim { it <= ' ' }
//            val passwordInput = edtxtPass.text.toString().trim { it <= ' ' }
//            val user: User = db.getUserByEmailAndPassword(userInput, passwordInput)
//            if (user == null) {
//                Toast.makeText(applicationContext, "Invalid Credentials", Toast.LENGTH_LONG).show()
//            } else {
//                LoggedInUser.setLoggedInUser(user)
//                val month: String
//                val year: String
//                month = getMonth(Calendar.getInstance()[Calendar.MONTH])
//                year = Calendar.getInstance()[Calendar.YEAR].toString() + ""
//                val budget =
//                    Budget(db.getCurrentBudget(LoggedInUser.getLoggedInUser().getID(), year, month))
//                LoggedInUser.setCurrentBudget(budget.getAmount())
//                if (LoggedInUser.getCurrentBudget().equals("")) {
//                    db.insertBudget(month, year, "0", LoggedInUser.getLoggedInUser().getID())
//                    LoggedInUser.setCurrentBudget("0")
//                } else {
//                    LoggedInUser.setCurrentBudget(
//                        db.getCurrentBudget(
//                            LoggedInUser.getLoggedInUser().getID(), year, month
//                        ).getAmount()
//                    )
//                }
//                finish()
//                startActivity(Intent(this@LoginActivity, NavigationDrawerActivity::class.java))
//            }
        }
    }
}